src/net/toxme.cpp

```diff
@@ -47,7 +45,38 @@ QByteArray Toxme::makeJsonRequest(QString json)
     return reply->readAll();
 }
 
-QByteArray Toxme::prepareEncryptedJson(int action, QString payload)
+QByteArray Toxme::getServerPubkey(QString url)
+{
+    // Get key
+    QNetworkAccessManager netman;
+    QNetworkRequest request{url};
+    request.setHeader(QNetworkRequest::ContentTypeHeader, "application/json");
+
+    QNetworkReply* reply = netman.get(request);
+    while (!reply->isFinished())
+        qApp->processEvents();
```

Comment:
This will make your head (and the processor) spin at 100%… :)

Try this:

```c++
while (reply->isRunning()) {
    reply->waitForReadyRead(100);
    qApp->processEvents();
}
```